_health-care/_js/components/HealthCareApp.jsx

```diff
@@ -215,7 +214,17 @@ class HealthCareApp extends React.Component {
   }
 
   handleContinue() {
-    hashHistory.push(this.getNextUrl());
+    const sectionPath = this.props.location.pathname.split('/').filter((path) => { return !!path; });
+    const sectionData = validations.initializeNullValues(_.get(this.state.applicationData, sectionPath));
+    this.publishStateChange(sectionPath, sectionData);
+
+    this.setState({}, () => {
+      if (validations.isValidSection(this.props.location.pathname, sectionData)) {
+        hashHistory.push(this.getNextUrl());
+      } else {
+        document.getElementsByClassName('usa-input-error')[0].scrollIntoView();
```

Comment:
Add a TODO here for me and maybe file an issue. I think this code may sometimes throw an exception.